### PR TITLE
Light and dark mode with toggle

### PR DESCRIPTION
**Pull request by AbhijeetSingla**
# About
Added dark and light mode with a toggle and local storage in javascript to save user preference.
# Features added
- Initial theme preference set
- Toggle between dark and light theme
- Used Local storage for setting user theme preference for next visit
# Preview
[Live Preview](https://theme4-as1553-component-library.netlify.app)

### CODE REVIEW

File: css/color-scheme.css [hunk 1]
```diff
@@ -130,6 +73,20 @@ body {
     font-size: 110%;
 }
 
+/* header */
+
+.header-base img {
+    display: block;
+    max-width: 500px;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css]: use rem for sizes greater than 16px

File: css/color-scheme.css [hunk 2]
```diff
@@ -379,6 +336,22 @@ body {
     align-items: center;
 }
 
+/* typography */
+
+.typography-base {
+    text-align: center;
+    flex-direction: column;
+    --gap: 0;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css]: i didn't understand what is this?

File: css/color-scheme.css [hunk 3]
```diff
@@ -441,6 +414,7 @@ body {
     align-items: center;
     gap: 1rem;
     line-height: 25px;
```
Comment (dev-ashishsethi) [inline css/color-scheme.css]: line height can make you go mad use cautiously

File: css/color-scheme.css [hunk 4]
```diff
@@ -455,6 +429,22 @@ body {
 button + button {
```
Comment (dev-ashishsethi) [inline css/color-scheme.css]: use class instead for better specificity

File: css/dark.css [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    --primary-link-text-color: var(--text-color);
+    --secondary-link-text-color: var(--primary-link-color);
+    --section-offwhite-color: var(--primary-base);
+    --gradient-color: var(--bg-color);
+    --white-base: #FFFFFF;
+    --white-smoke-base: #EBEBEB;
+    --black-base: #000000;
+    --grey-fade: #404040;
+    --hr-color: #303030;
+}
```
Comment (dev-ashishsethi) [inline css/dark.css:24]: naming is good

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

File: css/color-scheme.css (hunk 2 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css]: I used an utility class for flex, but wanted to add some custom properties to that, so defined those, also the gap variable is also defined in the utility class with a fallback of 1rem.

File: css/dark.css (hunk 5 above)
Comment (AbhijeetSingla) [reply css/dark.css:24]: # Thanks :)

File: css/color-scheme.css (hunk 1 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css]: This was to set the display container, so using pixels helps me visualize better.

File: css/color-scheme.css (hunk 3 above)
Comment (AbhijeetSingla) [reply css/color-scheme.css]: Noted :D

**Pull request merged**